pcsx2-qt/Settings/GameSummaryWidget.cpp

```diff
@@ -269,92 +270,129 @@ void GameSummaryWidget::populateTrackList(const GameList::Entry* entry)
 
 void GameSummaryWidget::onVerifyClicked()
 {
-	// Can't do this while a VM is running because of stupid CDVD.
+	// Check if a game is currently running
+	bool paused_vm = false;
 	if (QtHost::IsVMValid())
 	{
-		QMessageBox::critical(QtUtils::GetRootWidget(this), tr("Error"), tr("Cannot verify image while a game is running."));
-		return;
-	}
-
-	IsoHasher hasher;
-	Error error;
-	if (!hasher.Open(m_entry_path, &error))
-	{
-		setVerifyResult(QString::fromStdString(error.GetDescription()));
-		return;
-	}
+		const int result = QMessageBox::question(QtUtils::GetRootWidget(this), tr("Game Running"),
+			tr("A game is currently running. Would you like to pause the game while verifying the hash, or cancel verification?"),
+			QMessageBox::Ignore | QMessageBox::Cancel, QMessageBox::Cancel);
 
-	QtModalProgressCallback callback(this);
-	hasher.ComputeHashes(&callback);
-	if (callback.IsCancelled())
-		return;
+		if (result == QMessageBox::Cancel)
+			return;
 
-	const int hash_column = hasher.IsCD() ? 5 : 4;
-	int row = 0;
+		// User chose to pause - pause the VM
+		VMManager::SetPaused(true);
```

Comment:
Still getting crashes with this after resuming the game, specifically when the game being played is a different game to the one being hashed. I don't think this prompting to pause/unpause the game thing is a good idea, just throw up an error and be done with it.